### PR TITLE
Reviving applicationinsights source to deprecate Maven package

### PR DESCRIPTION
**Pull request by srnagar**
Fixes https://github.com/Azure/azure-sdk-for-java/issues/24264
Fixes https://github.com/Azure/azure-sdk-for-java/issues/27177

This PR revives an old [PR that removed Application Insights](https://github.com/Azure/azure-sdk-for-java/pull/8995). This library did have a [preview release](https://search.maven.org/artifact/com.microsoft.azure/azure-applicationinsights-query/1.0.0-Preview-1/jar) and should be marked as deprecated in favor of `azure-monitor-query`. 

So, to deprecate we have to do another release (beta.2).

### CODE REVIEW

File: eng/versioning/version_data.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 com.azure:azure-data-sdk-parent;1.3.0;1.3.0
 com.microsoft.azure:azure-batch;10.0.0;10.1.0-beta.1
-com.microsoft.azure:azure-applicationinsights-query;1.0.0-beta-2;1.0.0-beta-2
+com.microsoft.azure:azure-applicationinsights-query;1.0.0-Preview-1;1.0.0-Preview-2
```
Comment (srnagar) [inline eng/versioning/version_data.txt:7]: Using the same versioning scheme the first preview used to ensure users notice the updated version.
https://mvnrepository.com/artifact/com.microsoft.azure/azure-applicationinsights-query/1.0.0-Preview-1

**Review by scottaddie — APPROVED** [review]
Here are some suggestions to consider, and then this PR looks good.

File: sdk/applicationinsights/microsoft-azure-applicationinsights-query/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Azure Application Insights
```
Comment (scottaddie) [inline sdk/applicationinsights/microsoft-azure-applicationinsights-query/README.md:1]:
```suggestion
# Azure Application Insights Query client library for Java
```

File: sdk/applicationinsights/microsoft-azure-applicationinsights-query/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+> Please note, a newer package [com.azure:azure-monitor-query](https://search.maven.org/artifact/com.azure/azure-monitor-query)
+> for [Azure Application Insights Query](https://azure.microsoft.com/services/monitor/) is available as of October 2021. While this
+> package will continue to receive critical bug fixes and security fixes, we strongly encourage you to upgrade to the new package.
+> Read the [migration guide](https://aka.ms/azsdk/java/migrate/monitorquery) for more details.
```
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/applicationinsights/microsoft-azure-applicationinsights-query/README.md:6]: Done.

**Review by scottaddie — APPROVED** [review]

**Review by JimSuplizio — CHANGES_REQUESTED** [review]
If this is supposed to be a one-off release to deprecate, don't create a pipeline just for that, use the partner release pipeline. At this point, using the partner release pipeline will be on par with the previous releases. Part of the reason for this is that you're going to be using a non-standard version which will fail doing the version checks and version increment. I'd remove the ci.data.yml and the additional module in applicationinsights/pom.xml.

**Review by JimSuplizio — APPROVED** [review]
As per our conversation, don't create a pipeline for this. Build with the deprecated and release through the partner release pipeline.

**Pull request merged**